spring-cloud-aws-sqs/src/main/java/io/awspring/cloud/sqs/listener/SqsMessageListenerContainer.java

```diff
@@ -118,14 +118,21 @@ public SqsMessageListenerContainer(SqsAsyncClient sqsAsyncClient) {
 
 	@Override
 	protected Collection<ContainerComponentFactory<T, SqsContainerOptions>> createDefaultComponentFactories() {
-		Assert.isTrue(allQueuesSameType(),
-				"SqsMessageListenerContainer must contain either all FIFO or all Standard queues.");
+		assertAllQueuesSameType();
 		return Arrays.asList(new FifoSqsComponentFactory<>(), new StandardSqsComponentFactory<>());
 	}
 
-	private boolean allQueuesSameType() {
-		return getQueueNames().stream().allMatch(this::isFifoQueue)
-				|| getQueueNames().stream().noneMatch(this::isFifoQueue);
+	@Override
+	public void setQueueNames(Collection<String> queueNames) {
+		super.setQueueNames(queueNames);
+		assertAllQueuesSameType();
```

Comment:
I don't think we need to check this in two places, how about we only check it here on `setQueueNames` and do that before calling `super`?

Reply:
Done, and confirmed this still resolves the issue for me.